opentracing/tracer.py

```diff
@@ -37,16 +39,71 @@ class Tracer(object):
 
     _supported_formats = [Format.TEXT_MAP, Format.BINARY, Format.HTTP_HEADERS]
 
-    def __init__(self):
+    def __init__(self, scope_manager=ScopeManager()):
+        self._scope_manager = scope_manager
         self._noop_span_context = SpanContext()
         self._noop_span = Span(tracer=self, context=self._noop_span_context)
+        self._noop_scope = Scope(self._scope_manager, self._noop_span)
+
+    @property
+    def scope_manager(self):
+        """ScopeManager accessor"""
+        return self._scope_manager
+
+    def start_active_span(self,
+                          operation_name,
+                          finish_on_close,
```

Comment:
Follow-up for: https://github.com/opentracing/opentracing-python/pull/64#discussion_r165644789

Reply:
I think the `operation_name` must be mandatory, since I don't see a legit span where the name is `None`. A span should always represent a unit of work, and it must have a proper name I think.

For the `finish_on_close`, this is probably the most "safe" solution since we delegate implementers and developers to make their choice when instrumenting their code. The advantage of having a default, is that we cover the most common case (that I think is people using OpenTracing and not exactly Tracer implementers since they usually go with the advanced usage in any case).

In the case of having a default, anyone should have an opinion on that. I think that the most common case is a developer that may not be interested in having such fine-grained control in their code, while implementers have and that's why they can switch to `False` easily. We may check `opentracing-contrib` to have a sense of the usage, though this is still the implementer point of view and not the developers one.